sycl/plugins/unified_runtime/CMakeLists.txt

```diff
@@ -52,9 +84,30 @@ add_sycl_plugin(unified_runtime
     # These below belong to Unified Runtime PI Plugin only
     "pi_unified_runtime.hpp"
     "pi_unified_runtime.cpp"
-    LIBRARIES
+  LIBRARIES
     Threads::Threads
     UnifiedRuntime-Headers
+    UnifiedRuntimeLoader-Headers
+    UnifiedRuntimeLoader
     LevelZeroLoader-Headers
```

Comment:
I think we should link only with the UnifiedRuntimeLoader here. And the L0 adapter should be build as a standalone SHARED library that links with LevelZero loader. 

Something like this:
```
add_sycl_plugin(unified_runtime
  SOURCES
    "pi_unified_runtime.hpp"
    "pi_unified_runtime.cpp"
    LIBRARIES
    Threads::Threads
    UnifiedRuntimeLoader
    UnifiedRuntime-Headers
)
```

and:
```
add_library(ur_level_zero
    SHARED
        ${CMAKE_CURRENT_SOURCE_DIR}/ur_level_zero.cpp
        ${CMAKE_CURRENT_SOURCE_DIR}/ur_level_zero.hpp
        "../../../ur_bindings.hpp"
        "../../../pi2ur.hpp"
        "../../../pi2ur.cpp"
        "../../ur.hpp"
        "../../ur.cpp"
)

target_include_directories(ur_level_zero
  PRIVATE "${LEVEL_ZERO_INCLUDE_DIR}"
)

target_link_libraries(ur_level_zero
  PRIVATE
    LevelZeroLoader-Headers
    LevelZeroLoader
)
```

Reply:
ur_loader work to continue in https://github.com/intel/llvm/pull/8423